src/frontend/src/invites.tsx

```diff
@@ -17,95 +37,229 @@ export const Invites = () => {
         loadInvites();
     }, []);
 
+    const create = async () => {
+        setBusy(true);
+        const result = await window.api.call<any>(
+            "create_invite",
+            credits,
+            credits_per_user,
+            inviteRealm,
+        );
+        if ("Err" in result) alert(`Failed: ${result.Err}`);
+        else loadInvites();
+        setBusy(false);
+    };
+
+    const saveInvites = async () => {
+        for (const i in invites) {
+            const [code, invite] = invites[i];
+            if (!invite.dirty) continue;
+            const response = await window.api.call<any>(
+                "update_invite",
+                code,
+                invite.credits !== undefined && invite.credits >= 0
+                    ? invite.credits
+                    : null,
+                invite.realm_id !== undefined ? invite.realm_id : null,
+            );
+            if ("Err" in (response || {})) {
+                alert(`Error: ${response.Err}`);
+                setBusy(true);
+                await loadInvites(); // Set back to prior state
+                setBusy(false);
+                return;
+            }
+        }
+    };
+
+    const updateInvite = (
```

Comment:
@aligatorr89 I changed the way invites are updated (modify in-place and set the `dirty` flag).

This is because I uncovered a bug: whenever I would update credits, the realm id would be deleted because the previous code would not send the realm id as updated value.

Reply:
Ok, I noticed, and added to tests